### PR TITLE
Faster recompile

### PR DESCRIPTION
**Pull request by dschwoerer**
Especially nice if working with shared libraries, and only a single, small file needs to be recompiled.

Recompiling `bout++.cxx` can take a long time, and this issue issue is avoided this way.

Due to #1933 not useful for cmake ...

I accidentally included 73e7ad8 - but that might be useful anyway to avoid a regression?

### CODE REVIEW

File: src/bout++.cxx [hunk 1]
```diff
@@ -454,7 +455,7 @@ void printStartupHeader(int MYPE, int NPES) {
 #ifdef MD5SUM
   output_progress.write("MD5 checksum: {:s}\n", BUILDFLAG(MD5SUM));
 #endif
-  output_progress.write(_("Code compiled on {:s} at {:s}\n\n"), __DATE__, __TIME__);
+  output_progress.write(_("Code compiled on {:s} at {:s}\n\n"), boutcompiletime, boutcompiledate);
```
Comment (ZedThree) [inline src/bout++.cxx:458]: Typo:

```suggestion
  output_progress.write(_("Code compiled on {:s} at {:s}\n\n"), boutcompiledate, boutcompiletime);
```

**Review by ZedThree — CHANGES_REQUESTED** [review]
Nice, thanks @dschwoerer ! One small change please :)

**Comment by ZedThree** [comment]
The black test is failing because it's trying to run on everything and two files are not compatible with python 3.

**Comment by dschwoerer** [comment]
> The black test is failing because it's trying to run on everything and two files are not compatible with python 3.

I am confused why it is trying to do that now. Shouldn't black already be running on next?
I further thought that this has already succeded, where the files reintroduced accidentially?

**Comment by ZedThree** [comment]
black is failing on this file: https://github.com/boutproject/BOUT-dev/blob/d2750d5cd3fa7b17b75ffbc9001c9163839a2967/tools/tokamak_grids/pyGridGen/aeqdsk.py

which no longer exists in `next`. I think this is just because this branch was off an old version of `next`. I suggest merging in latest `next`

**Comment by dschwoerer** [comment]
It seems I forgot to fetch before merging next. Thanks for the catch, it now works :+1:

**Review by ZedThree — APPROVED** [review]

**Pull request merged**